tests/datastore.rs

```diff
@@ -183,6 +183,33 @@ mod datastore_tests {
         assert_eq!(fetched_events[0].data, e_diff_data.data);
     }
 
+    #[test]
+    fn test_event_replace() {
+        // Setup datastore
+        let ds = Datastore::new_in_memory();
+        let bucket = create_test_bucket(&ds);
+
+        // Insert event
+        let e1 = Event {
+            id: None,
+            timestamp: Utc::now(),
+            duration: Duration::seconds(0),
+            data: json_map!{"key": json!("value")}
+        };
+        ds.insert_events(&bucket.id, &[e1.clone()]).unwrap();
+
+        let mut fetched_events = ds.get_events(&bucket.id, None, None, None).unwrap();
+        assert_eq!(fetched_events.len(), 1);
+        let e2 = fetched_events.pop().unwrap();
+
+        ds.insert_events(&bucket.id, &[e2.clone()]).unwrap();
```

Comment:
You're not really modifying the event though?

Maybe try changing the duration and the data field?